plugins/api/discovery/src/org/apache/cloudstack/discovery/ApiDiscoveryServiceImpl.java

```diff
@@ -109,7 +107,12 @@ public boolean start() {
             }
             response.setRelated(responseName);
 
-            Field[] responseFields = apiCmdAnnotation.responseObject().getDeclaredFields();
+            List<Field> responseFields = new ArrayList<>(Arrays.asList(responseClass.getDeclaredFields()));
```

Comment:
Instead of doing this whole loop thing, what about using `org.reflections.ReflectionUtils.getAllFields(Class<?>)`? It will produce the same result with much less code.

Reply:
Sure, done